allauth/account/views.py

```diff
@@ -291,11 +293,23 @@ def get_context_data(self, **kwargs):
         return ret
 
     def get_initial(self):
-        if self.request.GET.get("email", None):
-            initial = super().get_initial()
-            initial["email"] = self.request.GET.get("email")
-            return initial
-        return super().get_initial()
+        intial = super(SignupView, self).get_initial()
+        twice_email = app_settings.SIGNUP_EMAIL_ENTER_TWICE
+        request_get = self.request.GET
+
+        def validate_and_set_email(email_key):
```

Comment:
This looks a little bit complex to me, e.g. there is no need to validate the email twice. How about simplifying into:

```
from django.core.validators import validate_email

def get_initial(self):
     intial = super().get_initial()
     email = self.request.GET("email")
     if email:
         try:
             email = validate_email(email)
         except ValidationError:
             pass
         else:
             intial["email"] = email
             if app_settings.SIGNUP_EMAIL_ENTER_TWICE:
                 intial["email2"] = email
     return intial
```